### ISSUE TITLE
Error while opening Docker Desktop. The stub received bad data. Error code: Wsl/Service/0x800706f7. provisioning docker wsl distros: setting main distro version: running wsl command "wsl.exe -d docker-desktop -u root -e wsl-bootstrap version set main-101757

### ISSUE THREAD

**Issue by AbdulAhad10** [microsoft/WSL#9879]
### Windows Version

Microsoft Windows [Version 10.0.22000.1761]

### WSL Version

WSL version: 1.1.6.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

2

### Repro Steps

Install WSL from the Windows Store
Launched WSL
Notice the error message in the console:
**Windows Subsystem for Linux has no installed distributions.

Use 'wsl.exe --list --online' to list available distributions
and 'wsl.exe --install <Distro>' to install.

Distributions can also be installed by visiting the Microsoft Store:
https://aka.ms/wslstore
Error code: Wsl/Service/CreateInstance/GetDefaultDistro/WSL_E_DEFAULT_DISTRO_NOT_FOUND
Press any key to continue...**

### Expected Behavior

While opening the Docker desktop it should launch and run without any error.

### Actual Behavior

While opening Docker Desktop it opens with an error 
![Screenshot 2023-03-30 180454](https://user-images.githubusercontent.com/61472512/229075991-27f90bd9-bd35-4770-8b4d-95adbf52a321.jpg)

**Comment by AbdulAhad10** [comment]
Hi @benhillis As per your comment I have created logs as per the steps provided above.
[WslLogs-2023-03-31_22-42-24.zip](https://github.com/microsoft/WSL/files/11124345/WslLogs-2023-03-31_22-42-24.zip)
Please guide me some solution to my problem.

**Comment by QAnders** [comment]
I have the exact same issue...

**Comment by AbdulAhad10** [comment]
> I have the exact same issue...

Hey, QAnders did you get any solution for this? I can't seem to find any as I have tried every solution present on google.

**Comment by QAnders** [comment]
@AbdulAhad10 nope, it's completely useless right now... I've had to add Ubuntu as dual boot to be able to work...
